locator/tablets.cc

```diff
@@ -430,6 +430,36 @@ class tablet_effective_replication_map : public effective_replication_map {
         return result;
     }
 
+    std::optional<tablet_routing_info> check_locality(const token& search_token) const override {
+        auto&& tablets = get_tablet_map();
+        auto tid = tablets.get_tablet_id(search_token);
+        auto&& info = tablets.get_tablet_info(tid);
+        auto host = get_token_metadata().get_my_id();
+        auto shard = this_shard_id();
+
+        auto make_tablet_routing_info = [&] {
+            auto token_range = std::make_pair(tablets.get_first_token(tid), dht::next_token(tablets.get_last_token(tid)));
```

Comment:
Better to use (a, b] rather than [a, b), because system.tablets uses the former so there's less chance for confusion.